### PR TITLE
[libffi] Use -llibffi on Windows

### PR DESCRIPTION
**Pull request by MidwestB**
Replace -lffi with -llibffi for Windows development, not just mingw

**Describe the pull request**

- #### What does your PR fix?  
  Gtkmm could not link in Visual Studio 19 on WIndows 10.  See #18328 for details

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  It should support all triplets, but only affect Windows.  I did not change the CI baseline.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  I think so.  This change allows the library to properly link in Windows when not using mingw.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  No, I don't think this qualifies as updating the entire port?

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

This is my first update to a project on Github.  If I've done something wrong, please let me know!

### CODE REVIEW

File: ports/libffi/portfile.cmake [hunk 1]
```diff
@@ -44,7 +44,7 @@ endif()
 
 vcpkg_copy_pdbs()
 vcpkg_fixup_cmake_targets()
-if(VCPKG_TARGET_IS_MINGW)
+if(VCPKG_TARGET_IS_WINDOWS)
```
Comment (NancyLi1013) [inline ports/libffi/portfile.cmake:47]: Could you please help confirm if this is also required on `VCPKG_TARGET_IS_MINGW`?

**Review by NancyLi1013 — COMMENTED** [review]
Could you please update "port-version": as 2 in vcpkg.json?

Also, please run `vcpkg x-add-version libtiff` to update version files when you commit all your changes.

Edit: It should be `vcpkg x-add-version libffi`, sorry, it's my fault.

**Comment by dg0yt** [comment]
@NancyLi1013 It is not `[libtiff]`.

File: ports/libffi/portfile.cmake (hunk 1 above)
Comment (MidwestB) [reply ports/libffi/portfile.cmake:47]: Is it possible to change the commit to VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_MINGW?  Or else do I make a new commit and try to merge it with this?

I suppose there could be an issue if it is compiled on Linux with Mingw as a target.

**Comment by MidwestB** [comment]
I think I have made the requested changes and committed them.  I haven't worked on a git project before - just trying to make this change as suggested to me in #18328.  

I wasn't sure where to run `vcpkg x-add-version libtiff` as I am making these edits on github rather than my own files.  I have updated the baseline and libffi json files with the versions.  If there is something else I am missing, please let me know. Classes are starting this week, so I may not have a lot of time to fix things.

Thank you!

File: ports/libffi/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/libffi/portfile.cmake]: `VCPKG_TARGET_IS_WINDOWS` includes `VCPKG_TARGET_IS_MINGW`. So this changed line means no change for MinGW.

**Comment by NancyLi1013** [comment]
> @NancyLi1013 It is not `[libtiff]`.

Thanks for your notice. @dg0yt It's my fault, It is `libffi`. I have updated the [comments](https://github.com/microsoft/vcpkg/pull/19655#pullrequestreview-734728686).

**Review by NancyLi1013 — COMMENTED** [review]
Please add `"port-version": 1,` below the following codes in vcpkg.json file.

https://github.com/microsoft/vcpkg/blob/master/ports/libffi/vcpkg.json#L3

Then run vpkg `x-add-version --overwrite libffi`.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
This makes sense to me, thanks for the PR @MidwestB!

**Pull request merged**